src/modules/crud.js

```diff
@@ -0,0 +1,49 @@
+const addScore = async (name, score, apiEndPoint, idLink, htmlElementToPostSuccesfulMessage) => {
+  const url = `${apiEndPoint}${idLink}scores/`;
+  const newScoreObject = {
+    user: name,
+    score,
+  };
+  let responseFromServer = '';
+  await fetch(url, {
+    method: 'POST',
+    body: JSON.stringify(newScoreObject),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  }).then((response) => response.json())
+    .then((data) => {
+      responseFromServer = data.result;
+    }).catch((error) => {
+      responseFromServer = error.message;
+    });
```

Comment:
- [ ] I think the idea behind using async / await syntax is to avoid using .then chaining or promises chaining which leads to something known as `chaining hell`. Here are a couple of examples of the old .then syntax vs the new async / await one.

`.then`:
``` 
fetch('/article/promise-chaining/user.json')
  // Load it as json
  .then(response => response.json())
  // Make a request to GitHub
  .then(user => fetch(`https://api.github.com/users/${user.name}`))
  // Load the response as json
  .then(response => response.json())
  // Show the avatar image (githubUser.avatar_url) for 3 seconds (maybe animate it)
  .then(githubUser => {
    let img = document.createElement('img');
    img.src = githubUser.avatar_url;
    img.className = "promise-avatar-example";
    document.body.append(img);

    setTimeout(() => img.remove(), 3000); // (*)
  });
```

`async/await`:
```
// read our JSON
  let response = await fetch('/article/promise-chaining/user.json');
  let user = await response.json();

  // read github user
  let githubResponse = await fetch(`https://api.github.com/users/${user.name}`);
  let githubUser = await githubResponse.json();

  // show the avatar
  let img = document.createElement('img');
  img.src = githubUser.avatar_url;
  img.className = "promise-avatar-example";
  document.body.append(img);

  // wait 3 seconds
  await new Promise((resolve, reject) => setTimeout(resolve, 3000));

  img.remove();

  return githubUser;
```

As you can see async/await it's a bit more readable. You might be wondering how can you catch the errors then. This is done through the try / catch blocks which are written as follows.
```
try {
  some async functionality
} catch(error) {
  handleError(error)
}
```
I hope you find this insightful and you're able to replace the `.then` and `.catch` on your code for pure `async / await` syntax.

[You can read more about it here.](https://javascript.info/async-await)

Reply:
Hey @D4nielJ thanks for the review. Weirdly I actually prefer using traditional promise chains. I seem to find it easier to catch errors but I'll use the async/ await recommended. It will be a good exercise anyway. Working on it now 👌